source/common/runtime/runtime_impl.cc

```diff
@@ -26,32 +23,57 @@ namespace Runtime {
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
 std::string RandomGeneratorImpl::uuid() {
-#ifdef __APPLE__
-  // TODO(zuercher): factor out a sensible interface for UUID generation to make it easier
-  // to put a faster implementation in later.
-  char generated_uuid[UUID_LENGTH + 1];
-  uuid_t uuid;
-  uuid_generate_random(uuid);
-  uuid_unparse(uuid, generated_uuid);
-  return std::string(generated_uuid);
-#else
-  int fd = open("/proc/sys/kernel/random/uuid", O_RDONLY);
-  if (-1 == fd) {
-    throw EnvoyException(fmt::format("unable to open uuid, errno: {}", strerror(errno)));
+  // Create UUID from Truly Random or Pseudo-Random Numbers.
+  // See: https://tools.ietf.org/html/rfc4122#section-4.4
+  uint8_t rand[16];
+  RAND_bytes(rand, 16);
```

Comment:
My thinking on this is to keep using real random, but actually do the following:

1) Store the random bytes in a `static thread_local` variable along with an iteration counter.
2) Select 2 bytes that we will sequence through in order.
3) Basically fill in 2 bytes with the sequence, meaning we will only need to refresh the random every ~64K UUIDs. (This is arbitrary, feel free to use less bits and make it like 16K or something).

This should further improve performance by several orders of magnitude and for the purposes that Envoy uses random UUIDs for I think will be just fine from a collision standpoint.